data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/impl/ReceiverVerticle.java

```diff
@@ -47,73 +51,118 @@
 
   private final HttpServerOptions httpServerOptions;
+  private final HttpServerOptions httpsServerOptions;
   private final Function<Vertx, IngressProducerReconcilableStore> ingressProducerStoreFactory;
   private final IngressRequestHandler ingressRequestHandler;
   private final ReceiverEnv env;
 
-  private HttpServer server;
+  private HttpServer httpServer;
+  private HttpServer httpsServer;
   private MessageConsumer<Object> messageConsumer;
   private IngressProducerReconcilableStore ingressProducerStore;
 
   public ReceiverVerticle(final ReceiverEnv env,
-                          final HttpServerOptions httpServerOptions,
-                          final Function<Vertx, IngressProducerReconcilableStore> ingressProducerStoreFactory,
-                          final IngressRequestHandler ingressRequestHandler) {
+      final HttpServerOptions httpServerOptions,
+      final HttpServerOptions httpsServerOptions,
+      final Function<Vertx, IngressProducerReconcilableStore> ingressProducerStoreFactory,
+      final IngressRequestHandler ingressRequestHandler) {
     Objects.requireNonNull(env);
     Objects.requireNonNull(ingressProducerStoreFactory);
     Objects.requireNonNull(ingressRequestHandler);
 
     this.env = env;
     this.httpServerOptions = httpServerOptions != null ? httpServerOptions : new HttpServerOptions();
+    this.httpsServerOptions = httpsServerOptions;
     this.ingressProducerStoreFactory = ingressProducerStoreFactory;
     this.ingressRequestHandler = ingressRequestHandler;
   }
 
-
   @Override
   public void start(final Promise<Void> startPromise) {
     this.ingressProducerStore = this.ingressProducerStoreFactory.apply(vertx);
     this.messageConsumer = ResourcesReconciler
-      .builder()
-      .watchIngress(IngressReconcilerListener.all(this.ingressProducerStore, this.ingressRequestHandler))
-      .buildAndListen(vertx);
+        .builder()
+        .watchIngress(IngressReconcilerListener.all(this.ingressProducerStore, this.ingressRequestHandler))
+        .buildAndListen(vertx);
+
+    this.httpServerOptions
+        .setSsl(false);
+
+    this.httpServer = vertx.createHttpServer(this.httpServerOptions);
+
+    // check whether the secret volume is mounted
+    File secretVolume = new File("/etc/receiver-secret-volume");
+    if (secretVolume.exists()) {
+      // The secret volume is mounted, we should start the https server
+      // check whether the tls.key and tls.crt files exist
+      File tlsKeyFile = new File("/etc/receiver-secret-volume/tls.key");
+      File tlsCrtFile = new File("/etc/receiver-secret-volume/tls.crt");
 
-    this.server = vertx.createHttpServer(httpServerOptions);
+      if (tlsKeyFile.exists() && tlsCrtFile.exists() && httpsServerOptions != null) {
+        PemKeyCertOptions keyCertOptions = new PemKeyCertOptions()
+            .setKeyPath("/etc/receiver-secret-volume/tls.key")
+            .setCertPath("/etc/receiver-secret-volume/tls.crt");
```

Comment:
Can we use final static class fields for various paths instead of duplicated them multiple times?